backend/src/utils/prometheusUtils.ts

```diff
@@ -43,7 +43,12 @@ export const callPrometheus = async (
             fastify.log.info('Successful response from Prometheus.');
             resolve({ code: 200, response: parsedData });
           } catch (e) {
-            fastify.log.error(`Failure parsing the response from Prometheus. ${e.message || e}`);
+            const errorMessage = e.message || e.toString();
+            fastify.log.error(`Failure parsing the response from Prometheus. ${errorMessage}`);
+            if (errorMessage.includes('Unexpected token < in JSON')) {
+              reject({ code: 422, response: 'Unprocessable prometheus response' });
+              return;
+            }
             fastify.log.error(`Unparsed Prometheus data. ${rawData}`);
             reject({ code: 500, response: rawData });
```

Comment:
Out of curiosity, do we ever really want to do a 500 here?

Reply:
The server ran into an error that shouldn't happen. It's a 500 by definition, no?

Reply:
well, I guess I was on the fence since in this case we were trying to look up something and failed to find it, perhaps we could have justified a 4XX.  I guess 500 or 502 is probably more appropriate.

Reply:
Well, the reason could be anything... but I suppose it's likely 502 in this case as the target of the call is misbehaving so it likely is having issues.